### ISSUE TITLE
Allow labels feature.node.kubernetes.io by cloud provider

### ISSUE THREAD

**Issue by sergelogvinov** [kubernetes/kubernetes#124769]
Hello

Can we add capability to set `feature.node.kubernetes.io/<feature-name>` by cloud provider?
I believe they was designed for that https://kubernetes.io/docs/reference/labels-annotations-taints/#feature-node-kubernetes-io

And probably `node.cloudprovider.kubernetes.io/` 

@mmerkes
Thank you.

https://github.com/kubernetes/kubernetes/blob/eeb42fa54306d1956b7df8d855b1e8219c9f9a53/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L545-L558

**Comment by elmiko** [comment]
discussing this at the sig cloud provider meeting today, we have a question about if using this label will cause any disruption with node feature discovery component.

we need to be careful about colliding with other components that operate in the `kubernetes.io` namespace. we should reach out to the NFD maintainers to understand if there are any guidelines around using that label.

we are accepting this triage now, but would like to see followup with the NFD maintainers to better understand the problem and possible collisions with the labels.

/triage accepted

**Comment by elmiko** [comment]
this seems like a worthwhile question, but i'm not sure about the possible disruption to NFD.

/triage accepted

**Comment by elmiko** [comment]
i don't think we have answered the question about if this would clash with NFD. i am going to leave this as untriaged so that the SIG will discuss it again.

**Comment by elmiko** [comment]
we are discussing this in the SIG Cloud Provider meeting today. although this is about node features of the hypervisor, the current workaround that @sergelogvinov is using is based on reading the instance type from the cloud provider to know about the features. the other complicating factor is that the information is available from the linux kernel and that is where it is being harvested from in some cases (for example hugepages).

we'd like to get some input from sig-node on how this might operate with the node feature discovery operator.

/sig node

**Comment by elmiko** [comment]
we are discussing this issue in our triage, we think at this point it would be worthwhile to take this issue to the sig node meeting if we want to get more traction on it.

https://github.com/kubernetes/community/blob/main/sig-node/README.md#meetings

**Comment by sergelogvinov** [comment]
Thank you! 

It probably makes sense to close this issue for now and open a new one once I have a better understanding of how `node-feature-discovery` works under the hood.

**Comment by elmiko** [comment]
based on @sergelogvinov 's suggestion, i am closing this. i welcome a followup issue given the new understanding, thanks again!
